### PR TITLE
chore(submitter): remove dumpPrivKey

### PR DESCRIPTION
**Pull request by Lazar955**
This pull request removes the usage of the dumpPrivKey RPC method from the submitter and end-to-end (E2E) tests. The dumpPrivKey method is specific to legacy wallets and is no longer supported in the latest versions of bitcoind. Eliminating its usage is a necessary step for upgrading to newer versions of bitcoind, as its deprecation would otherwise hinder the upgrade process.

Updates the version of bitcoind to v27.0

References [issue](https://github.com/babylonchain/vigilante/issues/227).

### CODE REVIEW

File: e2etest/bitcoind_node_setup.go
Comment (Lazar955) [inline e2etest/bitcoind_node_setup.go:96]: Legacy wallets were "false"

File: submitter/relayer/utils.go
Comment (Lazar955) [inline submitter/relayer/utils.go:28]: When using fundRawTx in out txs sometimes changes the order, so we need to find the one with change address

File: e2etest/bitcoind_node_setup.go
Comment (Lazar955) [inline e2etest/bitcoind_node_setup.go:97]: Legacy wallets were "false", true for descriptor wallets

File: submitter/relayer/relayer.go
Comment (Lazar955) [inline submitter/relayer/relayer.go:28]: size of the change address in txOut

File: e2etest/test_manager.go
Comment (Lazar955) [inline e2etest/test_manager.go:258]: A bit of docs on the topic:
https://bips.dev/384/
https://github.com/bitcoin/bitcoin/blob/master/doc/descriptors.md

Basically `combo` allows us to import a key and handle multiple types of btc scripts with a single descriptor command.

 "now" as a timestamp tells Bitcoind to start scanning from the current blockchain height

File: submitter/relayer/relayer.go
Comment (KonradStaniec) [inline submitter/relayer/relayer.go:400]: commented out code

File: submitter/relayer/relayer.go
Comment (KonradStaniec) [inline submitter/relayer/relayer.go:398]: While looking at this function docs - https://developer.bitcoin.org/reference/rpc/fundrawtransaction.html#argument-2-options
it seems we can specify the position of change output in arguments - `changePosition`.

Maybe it would be better to use this option instead finding change output on our own ?

File: e2etest/test_manager.go
Comment (KonradStaniec) [reply e2etest/test_manager.go:258]: I would add this line with brief description above the line where we create descriptor

File: submitter/relayer/relayer.go
Comment (KonradStaniec) [inline submitter/relayer/relayer.go:276]: unnecessary comment ?

File: submitter/relayer/relayer.go
Comment (KonradStaniec) [inline submitter/relayer/relayer.go:389]: the `utxo *types.UTXO` argument to the function is probably not needed now as all intput to the transaction as selected by `rl.BTCWallet.FundRawTransaction` function.

Given that we probably need two versions of this function:
1. first version for the first transaction, in which we allow wallet to chose inputs automatically
2. second version for the second transaction, in which we are able to specifiy input manually to use the transaction chaining

File: e2etest/test_manager.go
Comment (KonradStaniec) [inline e2etest/test_manager.go:147]: commented out code

**Review by RafilxTenfen — COMMENTED** [review]
PR looks good, but what about [dumpPrivKeyAndSignTx](https://github.com/babylonlabs-io/vigilante/blob/d73b523c409a1444bb0ea94cf3dfa299d1a6e27a/submitter/relayer/relayer.go#L440) at the `processCheckpoints` case

**Review by RafilxTenfen — APPROVED** [review]
Code changes look good to me ^^

File: submitter/relayer/relayer.go
Comment (gitferry) [reply submitter/relayer/relayer.go:389]: Not sure how `FundRawTransaction` would simplify things as we cannot control the inputs used by `FundRawTransaction`. As such, for (1), how do we know the value of the output is sufficient to fund the second tx for chaining?

**Review by gitferry — COMMENTED** [review]
Signing part lgtm! As per off-line discussion, my major concern is whether `FundRawTransaction` can handle the chaining without introducing much complexity. Maybe this part can be improved in a separate pr to not block here

File: submitter/relayer/relayer.go
Comment (KonradStaniec) [reply submitter/relayer/relayer.go:389]: If I understand our chaining mechanism correctly, want we want to do is:
1. send first transaction with pretty low fee rate, lets call it `Tx1`
2. then send the second transaction (`Tx2`)with higher fee rate and with input that is `Tx1` change output (this is necessary to achieve the chaining)

Imo `FundRawTransaction` can help here a lot and simplify stuff as:
1. For the `Tx1` we can just create unsigned and unfunded transaction with one output, and send it to `FundRawTransaction`, it will automatically choses inputs and add one change output which will have correct value to obey specified fee rate.
2. Then we can create `Tx2` with one input (change output from `Tx1` ) and one output and send it to `FundRawTransaction`. Now based on specified fee rate, endpoint will attach or not other inputs and will attach one change output to `Tx2` . This way chaining will be preserved,  and second tx will obey specified fee rate

Comment (gitferry) [reply submitter/relayer/relayer.go:389]: Oh, I didn't realize in (2) we can inject one input from the change output of tx1. If the case, then it indeed improves a lot. Sounds good to me!

**Review by KonradStaniec — APPROVED** [review]
We minor nits. Before merging lets wait for @gitferry approval as he is most familiar with this logic.

File: submitter/relayer/relayer.go
Comment (gitferry) [inline submitter/relayer/relayer.go:293]:
```suggestion
// ChainTwoTxAndSend builds two chaining txs with the given data
// the second tx consumes the output of the first tx
```

File: submitter/relayer/relayer.go
Comment (gitferry) [inline submitter/relayer/relayer.go:162]: why we don't want to estimate the fee here?

File: submitter/relayer/relayer.go
Comment (gitferry) [inline submitter/relayer/relayer.go:169]: This seems outdated

File: submitter/relayer/relayer.go
Comment (gitferry) [inline submitter/relayer/relayer.go:172]: can we make it an issue for reminder?

File: submitter/relayer/relayer.go
Comment (gitferry) [inline submitter/relayer/relayer.go:327]: this doc string needs update

File: submitter/relayer/relayer.go
Comment (gitferry) [inline submitter/relayer/relayer.go:333]: can we make `1` a self-explained constant?

File: submitter/relayer/relayer.go
Comment (gitferry) [inline submitter/relayer/relayer.go:410]: should we also check the change is higher than dust?

File: submitter/relayer/relayer.go
Comment (gitferry) [inline submitter/relayer/relayer.go:412]: We probably need to remove the balance of input as we don't specify it anymore. And we need to make it a single line as otherwise the values are not well translated

**Review by gitferry — COMMENTED** [review]
Sorry for the delay. Some quick comments:

File: submitter/relayer/relayer.go
Comment (Lazar955) [reply submitter/relayer/relayer.go:162]: We've changed how we estimate fee, by using the fee value from the response of `FundRawTransaction`. This produces a different value (a bit higher fee) than we used to, which means when we estimate the fee here it wouldn't be the same way as we did, and in local tests resulting in a lower calculated `bumpedFee` needing a higher `ResubmitFeeMultiplier`. So I see it either as estimating fee ourselves or using the response from `FundRawTransaction`. Open to any other inputs, I might be missing something

Comment (gitferry) [reply submitter/relayer/relayer.go:162]: This makes sense. Thanks for the explanation!

**Pull request merged**
